models/discounted-price/src/builders.spec.ts

```diff
@@ -0,0 +1,39 @@
+import { DiscountedPriceRest, DiscountedPriceGraphql } from './index';
+
+describe('DiscountedPrice Builder', () => {
+  it('should build properties for the discounted price REST representation', () => {
+    const restModel = DiscountedPriceRest.random().build();
+
+    expect(restModel).toEqual(
+      expect.objectContaining({
+        value: expect.objectContaining({
+          id: expect.any(String),
+          value: expect.any(Object),
+        }),
+        discount: expect.objectContaining({
+          id: expect.any(String),
+        }),
+      })
+    );
+  });
+
+  it('should build properties for the discounted price GraphQL representation', () => {
+    const graphqlModel = DiscountedPriceGraphql.random().build();
+
+    expect(graphqlModel).toEqual(
+      expect.objectContaining({
+        __typename: 'DiscountedProductPriceValue',
+        value: expect.objectContaining({
+          id: expect.any(String),
+          value: expect.any(Object),
+        }),
```

Comment:
```suggestion
        value: expect.objectContaining({
          id: expect.any(String),
          value: expect.any(Object),
          __typename: 'ProductPrice',
        }),
```

Reply:
In the Graphql representations is important to validate the `__typename` attribute as that's what graphql clients use to know about the types in a response.